### PR TITLE
circumvent cache look up for OBO

### PR DESCRIPTION
**Pull request by jennyf19**
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1349

### CODE REVIEW

File: src/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs [hunk 1]
```diff
@@ -29,20 +28,7 @@ internal class OnBehalfOfRequest : RequestBase
...
-
-            MsalAccessTokenCacheItem msalAccessTokenItem = await CacheManager.FindAccessTokenAsync().ConfigureAwait(false);
-            if (msalAccessTokenItem != null)
-            {
-                return new AuthenticationResult(msalAccessTokenItem, null, AuthenticationRequestParameters.RequestContext.CorrelationId);
-            }
-
-
+            
             var msalTokenResponse = await SendTokenRequestAsync(GetBodyParameters(), cancellationToken).ConfigureAwait(false);
```
Comment (jmprieur) [inline src/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs]: Do I understand correctly that:
1. this is now applying the recommended pattern?
2. But this is a breaking change for customers who were not applying this pattern?

Comment (bgavrilMS) [reply src/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs]: Yes!

File: src/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs [hunk 2]
```diff
@@ -29,20 +28,7 @@ internal class OnBehalfOfRequest : RequestBase
         internal override async Task<AuthenticationResult> ExecuteAsync(CancellationToken cancellationToken)
         {
             await ResolveAuthorityEndpointsAsync().ConfigureAwait(false);
-
-            // look for access token in the cache first.
-            // no access token is found, then it means token does not exist
-            // or new assertion has been passed. We should not use Refresh Token
-            // for the user because the new incoming token may have updated claims
-            // like mfa etc.
```
Comment (bgavrilMS) [inline src/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs]: @jennyf19  - do you understand this comment? The caching logic here was added for a purpose, but seems to be forgotten.

File: tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs [hunk 3]
```diff
@@ -173,7 +173,9 @@ public async Task JsonWebTokenWithX509PublicCertSendCertificateOnBehalfOfTestAsy
                 Assert.IsNotNull(result.AccessToken);
 
                 appCacheAccess.AssertAccessCounts(0, 0);
-                userCacheAccess.AssertAccessCounts(1, 1);
+                userCacheAccess.AssertAccessCounts(0, 1);
+
+                harness.HttpManager.AddMockHandler(CreateTokenResponseHttpHandler(false));
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs]: There are OBO tests that are commented. I think this is a good oportunity to bring them back to life.

File: src/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs (hunk 2 above)
Comment (jmprieur) [reply src/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs]: Would it mean that our guidance is wrong?

File: tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs (hunk 3 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs]: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/bd8b7f693a01a058322198d4deb782e3eba831fe/tests/Microsoft.Identity.Test.Unit.net45/RequestsTests/OboFlowTests.cs#L17

Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs]: I think they were commented out when we moved to common core or smth. So they are pretty out of date possibly.

**Pull request closed without merging**